### PR TITLE
is reRefactor/define class options at top of file

### LINKED ISSUE #149: classOptions should be defined at the top of the file above the class
And used with `mergeOptions` instead of pick

This is nice self documenting code and `App` is already part of the way there.

We'll want to do this for anything that merge's options onto `this`

We should also remove all instances of `getOption` and just use `this.optionName` when it gets merged.

### PR DESCRIPTION
**Pull request by TerrenceLJones**
This resolves #149.

### CODE REVIEW

**Comment by paulfalgout** [comment]
Ah so the `getOption` part of this is great.

I actually think it might be better not to define those defaults on the class... just because we want people to pass in things via the options or to override the value..  But we _don't_ want people to add to an existing hash..

What I meant in #149 was taking the options merged in the constructor and defining them in an array outside of the class at the top of the module.  So like:
```
const classOptions = [
  'viewEventPrefix', 
  'ViewClass', 
  'viewOptions', 
  'region'
];

const Component = Marionette.Object.extend({
...
  constructor(options) {
    this.mergeOptions(options, classOptions);
...
  }

```

This pattern is kind of taken from backbone, but I think it might look nicer in the modules to define them at the top of the file, but I'm certainly open to criticism there.
https://github.com/jashkenas/backbone/blob/master/backbone.js#L1245

I'm also on the fence about `classOptions` vs `ClassOptions` vs `CLASS_OPTIONS` etc etc

Basically if you're reading through the source at the top of each module it defines all of the options available throughout the class before you dig into the inner-workings.

**Comment by TerrenceLJones** [comment]
@paulfalgout Ah that makes sense...and should be a quick fix. I would suggest maybe `ClassOptions`? The `CLASS_OPTIONS` seems to constant'ish and the capital *C* **seems** more appropriate.

File: src/app.js [hunk 1]
```diff
@@ -4,6 +4,13 @@ import StateMixin from './mixins/state';
 import ChildAppsMixin from './mixins/child-apps';
 import EventListenersMixin from './mixins/event-listeners';
 
+const pickOptions = [
```
Comment (paulfalgout) [inline src/app.js]: I think they should all be called the same thing.  `ClassOptions` works for me.

Comment (TerrenceLJones) [inline src/app.js]: @paulfalgout Crap...good catch...they were suppose to be the same name.
